src/test/java/com/github/princesslana/eriscasper/repository/event/TestGuildsFromEvents.java

```diff
@@ -0,0 +1,56 @@
+package com.github.princesslana.eriscasper.repository.event;
+
+import com.github.princesslana.eriscasper.data.event.ChannelCreateEvent;
+import com.github.princesslana.eriscasper.data.event.ChannelDeleteEvent;
+import com.github.princesslana.eriscasper.data.event.Event;
+import com.github.princesslana.eriscasper.data.event.GuildCreateEvent;
+import com.github.princesslana.eriscasper.data.event.GuildDeleteEvent;
+import com.github.princesslana.eriscasper.data.resource.Channel;
+import com.github.princesslana.eriscasper.data.resource.Guild;
+import com.github.princesslana.eriscasper.faker.DataFaker;
+import io.reactivex.subjects.PublishSubject;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class TestGuildsFromEvents {
+
+  private PublishSubject<Event> events = PublishSubject.create();
+
+  private GuildsFromEvents subject;
+
+  @BeforeMethod
+  public void subject() {
+    subject = new GuildsFromEvents(events);
+  }
+
+  @Test
+  public void getGuild_whenAfterGuildEvent_shouldCacheGuild() {
+    Guild guild = DataFaker.guild();
+    Channel channel = guild.getChannels().get(0);
+    events.onNext(GuildCreateEvent.of(guild));
+    Guild n = DataFaker.guild();
+    events.onNext(GuildCreateEvent.of(n));
+    subject.getGuild(n.getId()).test().assertValue(n);
+    subject.getGuild(guild.getId()).test().assertValue(guild);
```

Comment:
Can we add a comment here as to why we're calling this many times?

Or, break it out into a second test `getGuild_whenGetGuildCalledTwice_shouldReturnSameGuild` ?